pkg/pillar/zedUpload/azureutil/azure.go

```
@@ -242,3 +244,91 @@ func GetAzureBlobMetaData(accountName, accountKey, containerName, remoteFile str
 	stringHex := hex.EncodeToString(decodedString)
 	return blob.Properties.ContentLength, stringHex, nil
 }
+
+// GenerateBlobSasURI is used to generate the URI which can be used to access the blob until the the URI expries
+func GenerateBlobSasURI(accountName, accountKey, containerName, remoteFile string, httpClient *http.Client) (string, error) {
+	c, err := NewClient(accountName, accountKey, httpClient)
+	if err != nil {
+		return "", err
+	}
+	blobClient := c.GetBlobService()
+	container := blobClient.GetContainerReference(containerName)
+	containerExists, _ := container.Exists()
+	if !containerExists {
+		return "", err
```

If container doesn't exist. Please return a proper error saying "Container doesn't exist".